src/pkg/repository/repository.go

```diff
@@ -204,53 +169,18 @@ type ConnConfig struct {
 //   - connect to the m365 account
 //   - connect to the provider storage
 //   - return the connected repository
-func (r *repository) Connect(
-	ctx context.Context,
-	cfg ConnConfig,
-) (err error) {
-	ctx = clues.Add(
-		ctx,
-		"acct_provider", r.Account.Provider.String(),
-		"acct_id", clues.Hide(r.Account.ID()),
-		"storage_provider", r.Storage.Provider.String())
-
-	defer func() {
-		if crErr := crash.Recovery(ctx, recover(), "repo connect"); crErr != nil {
-			err = crErr
-		}
-	}()
+func (r *repository) Connect(ctx context.Context, cfg ConnConfig) (err error) {
+	ctx = r.addContextInfo(ctx)
+	defer r.handleRecovery(ctx, &err, "repo connect")
 
 	if err := r.ConnectDataProvider(ctx, cfg.Service); err != nil {
 		return clues.Stack(err)
 	}
 
 	observe.Message(ctx, "Connecting to repository")
 
-	kopiaRef := kopia.NewConn(r.Storage)
-	if err := kopiaRef.Connect(ctx, r.Opts.Repo); err != nil {
-		return clues.Wrap(err, "connecting kopia client")
-	}
-	// kopiaRef comes with a count of 1 and NewWrapper/NewModelStore bumps it again so safe
-	// to close here.
-	defer kopiaRef.Close(ctx)
-
-	r.dataLayer, err = kopia.NewWrapper(kopiaRef)
-	if err != nil {
-		return clues.Stack(err).WithClues(ctx)
-	}
-
-	r.modelStore, err = kopia.NewModelStore(kopiaRef)
-	if err != nil {
-		return clues.Stack(err).WithClues(ctx)
-	}
-
-	if r.ID == events.RepoIDNotFound {
-		rm, err := getRepoModel(ctx, r.modelStore)
-		if err != nil {
-			return clues.Wrap(err, "retrieving repo model info")
-		}
-
-		r.ID = string(rm.ID)
+	if err := r.setupKopia(ctx, ctrlRepo.Retention{}, false); err != nil {
+		return err
```

Comment:
Always stack errors, please
```suggestion
		return clues.Stack(err)
```